src/Entities/Misc/TimedTriggerSpikes.cs

```diff
@@ -312,6 +345,11 @@ public TimedTriggerSpikes(Vector2 position, int size, Directions direction, stri
         }
 
         public override void Update() {
+            if (rainbow) {
+                if (crystalSpinner == null) { crystalSpinner = new CrystalStaticSpinner(Vector2.One * -1000 + offset, false, CrystalColor.Rainbow); }
+                if (Scene == null) { return; }
+                if (!Scene.Tracker.GetEntities<CrystalStaticSpinner>().Contains(crystalSpinner)) { Scene.Add(crystalSpinner); }
+            }
```

Comment:
This could definitely be improved. 
Under what circumstances would `crystalSpinner` be `null` at this point? the same for `Scene`, since the entity should then no longer be updated.
If you want to check if `crystalSpinner` is part of the current scene it's probably best to do `crystalSpinner.Scene == Scene`.